INTEGRATION.md

```diff
@@ -0,0 +1,307 @@
+# Integration guide
+
+Deployment addresses can be found at:
+
+- Mainnet: [mainnet-deployment-info.json](mainnet-deployment-info.json)
+- Testnet: [testnet-deployment-info.json](testnet-deployment-info.json)
+
+## Ethereum
+
+Liquid staking is achieved through `MaticX` contract and the yield-bearing ERC-20 token `MaticX` is given to the user.
+
+### 1. Stake Matic on Ethereum
+
+Send Matic and receive liquid staking MaticX token.  
+_MaticX approval should be given prior._
+
+```SOLIDITY
+IMatic matic = IMatic(MATIC_ADDRESS);
+IMaticX maticX = IMaticX(MATICX_ADDRESS);
+require(matic.approve(MATICX_ADDRESS, _amountInMatic), "Not approved");
+uint256 amountInMaticX = maticX.submit(_amountInMatic);
+
+emit StakeEvent(msg.sender, msg.value, amountInMaticX);
+```
+
+### 2. Unstake Matic on Ethereum
+
+Send MaticX and create a withdrawal request.  
+_MaticX approval should be given prior._
+
+```SOLIDITY
+IMaticX maticX = IMaticX(MATICX_ADDRESS);
+require(
+    maticX.approve(MATICX_ADDRESS, _amountInMaticX),
+    "Not approved"
+);
+maticX.requestWithdraw(_amountInMaticX);
+
+emit UnstakeEvent(msg.sender, _amountInMaticX);
+```
+
+### 3. Claim Matic on Ethereum
+
+After 3-4 days (80 checkpoints), Matic can be withdrawn.
+
+```SOLIDITY
+IMatic matic = IMatic(MATIC_ADDRESS);
+IMaticX maticX = IMaticX(MATICX_ADDRESS);
+
+// Claim all available withdrawal requests
+WithdrawalRequest[] memory requests = getUserWithdrawalRequests(
+    msg.sender
+);
+
+// StakeManager is necessary to check the availability of the withdrawal request
+IStakeManager stakeManager = IStakeManager(STAKEMANAGER_ADDRESS);
+for (uint256 idx = requests.length - 1; idx >= 0; idx--) {
```

Comment:
Should we mention in the comments, that it is very important array is read in reverse, else there will be errors.

Reply:
Good idea. Added a comment on it. https://github.com/stader-labs/maticX/pull/34/commits/60a6678f66369d326e4e96cdac72d1696abd8b43